tests/v2/utils/library/test_utils.go

```diff
@@ -299,37 +299,97 @@ func verify(devfile *commonUtils.TestDevfile) error {
 // editCommands modifies random attributes for each of the commands in the devfile.
 func editCommands(devfile *commonUtils.TestDevfile) error {
 
+	var errorString []string
 	commonUtils.LogInfoMessage(fmt.Sprintf("Edit %s : ", devfile.FileName))
 
 	commonUtils.LogInfoMessage(fmt.Sprintf(" -> Get commands %s : ", devfile.FileName))
 	commands, err := devfile.Follower.(DevfileFollower).LibraryData.GetCommands(common.DevfileOptions{})
 	if err != nil {
-		commonUtils.LogErrorMessage(fmt.Sprintf("Getting commands from library : %s : %v", devfile.FileName, err))
+		errorString = append(errorString, commonUtils.LogErrorMessage(fmt.Sprintf("Getting commands from library : %s : %v", devfile.FileName, err)))
+	} else if len(commands) < 1 {
+		errorString = append(errorString, commonUtils.LogErrorMessage("Updating commands : No commands returned"))
 	} else {
 		for _, command := range commands {
 			err = UpdateCommand(devfile, command.Id)
 			if err != nil {
-				commonUtils.LogErrorMessage(fmt.Sprintf("Updating command : %s : %v", devfile.FileName, err))
+				errorString = append(errorString, commonUtils.LogErrorMessage(fmt.Sprintf("Updating command : %s : %v", devfile.FileName, err)))
 			}
 		}
 	}
+	if len(errorString) > 0 {
+		err = errors.New(fmt.Sprint(errorString))
+	}
 	return err
 }
 
 // editComponents modifies random attributes for each of the components in the devfile.
 func editComponents(devfile *commonUtils.TestDevfile) error {
 
+	var errorString []string
 	commonUtils.LogInfoMessage(fmt.Sprintf("Edit %s : ", devfile.FileName))
 
 	commonUtils.LogInfoMessage(fmt.Sprintf(" -> Get components %s : ", devfile.FileName))
 	components, err := devfile.Follower.(DevfileFollower).LibraryData.GetComponents(common.DevfileOptions{})
 	if err != nil {
-		commonUtils.LogErrorMessage(fmt.Sprintf("Getting components from library : %s : %v", devfile.FileName, err))
+		errorString = append(errorString, commonUtils.LogErrorMessage(fmt.Sprintf("Getting components from library : %s : %v", devfile.FileName, err)))
+	} else if len(components) < 1 {
+		errorString = append(errorString, commonUtils.LogErrorMessage("Updating components : No components returned"))
 	} else {
 		for _, component := range components {
 			err = UpdateComponent(devfile, component.Name)
 			if err != nil {
-				commonUtils.LogErrorMessage(fmt.Sprintf("Updating component : %s : %v", devfile.FileName, err))
+				errorString = append(errorString, commonUtils.LogErrorMessage(fmt.Sprintf("Updating component : %s : %v", devfile.FileName, err)))
+			}
+		}
+	}
+	if len(errorString) > 0 {
+		err = errors.New(fmt.Sprint(errorString))
+	}
+	return err
+}
+
+// editProjects modifies random attributes for each of the projects in the devfile.
+func editProjects(devfile *commonUtils.TestDevfile) error {
```

Comment:
shouldnt this be called from `RunTest`

Reply:
Yes, lol, not sure how I lost that code. Too many branches!